### DISCUSSION TITLE
How do I configure basedpyright?

### DISCUSSION THREAD

**Discussion by typecasto** [LazyVim/LazyVim#3350] [Q&A] [answered]
The type errors aren't too helpful to me, so I'd like to set `typeCheckingMode` to `"standard"`. How would I do this? I know how to do it in pyproject.toml, but I'd like to set this as an editor setting so that I don't have to keep setting it in every workspace. 

My current config (in lua/plugins/violet.lua) is this:
```lua
  {
    "neovim/nvim-lspconfig",
    opts = {
      inlay_hints = { enabled = false },
    },
    servers = { basedpyright = { typeCheckingMode = "standard" } },
    setup = {
      basedpyright = function()
        require("lspconfig").basedpyright.setup({
          settings = {
            typeCheckingMode = "standard",
          },
        })
      end,
    },
  },

```
I've tried both of these methods individually (meaning just `servers`, or just `setup`), neither of them work.

**Comment by dpetka2001** [comment]
I think it should be
```lua
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        basedpyright = {
          settings = {
            basedpyright = {
              typeCheckingMode = "standard",
            },
          },
        },
      },
    },
  },
```
But I don't use it, so not 100% sure.

**Reply by typecasto** [reply]
Doesn't work, either that's not right, or maybe basedpyright just doesn't support being configured through LSP like this.

**Reply by dpetka2001** [reply, marked as answer]
It should be 
```lua
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        basedpyright = {
          settings = {
            basedpyright = {
			  analysis = {
              typeCheckingMode = "standard",
 			  },
            },
          },
        },
      },
    },
  },
```
I missed the `analysis` section in my previous post.

**Reply by typecasto** [reply]
That also doesn't seem to work.

**Reply by dpetka2001** [reply]
Are you sure those type errors come from basedpyright and not another tool?

**Reply by typecasto** [reply]
Ah, my mistake was assuming running `:LspRestart` would reload the config. I restarted neovim and now it works. Thanks!

**Reply by Act0r1** [reply]
can u say, how i can turn on inlay hints ?

**Reply by Dronakurl** [reply]
`<leader>uh`

Also, maybe you missed the option (put that in `options.lua`)
`vim.g.lazyvim_python_lsp = "basedpyright"`

**Comment by folke** [comment]
You can create a `pyproject.toml` file with

```toml
[tool.basedpyright]
typeCheckingMode = "standard"
```

**Reply by typecasto** [reply]
Is there somewhere I can store this that it will always take effect? I don't want to have to have one of these for every script I want to edit.

**Comment by Dronakurl** [comment]
I just discoverd the lang.python extra for myself.
It also has an option for basedpyright
just set 
`vim.g.lazyvim_python_lsp = "basedpyright"`
See also here:
https://github.com/LazyVim/LazyVim/blob/fa2fc6403185faabc47741b3aa2809192870b92e/lua/lazyvim/plugins/extras/lang/python.lua#L59

**Comment by kanishkaverma** [comment]
```            vim.lsp.config['basedpyright'] = {

              settings = {
                basedpyright = {
                  analysis = {
                    autoSearchPaths = true,
                    typeCheckingMode = 'none',
                    useLibraryCodeForTypes = true,
                    diagnosticMode = 'workspace',
                  },
                },
              },
            }

            vim.lsp.enable 'basedpyright'
```

even with this the type warnings are not going away any fix?

**Reply by Jacobus-afk** [reply]
typeCheckingMode = "off",

**Reply by Dronakurl** [reply]
https://github.com/microsoft/pyright/blob/main/docs/configuration.md#type-check-diagnostics-settings

**Reply by Dronakurl** [reply]
(see also: https://docs.basedpyright.com/dev/benefits-over-pyright/better-defaults/#typecheckingmode)
I had my problems with the typecheckingmode when switching from pyright to basedpyright.
Here is my config I ended up with:

```lua
opts = {
        inlay_hints = { enabled = false },
        servers = {
          basedpyright = {
            settings = {
              basedpyright = {
                analysis = {
                  ignorePatterns = { "*.pyi" },
                  diagnosticSeverityOverrides = {
                    reportCallIssue = "warning",
                    reportUnreachable = "warning",
                    reportUnusedImport = "none",
                    reportUnusedCoroutine = "warning",
                  },
                  -- diagnosticMode = "workspace",
                  diagnosticMode = "openFilesOnly",
                  typeCheckingMode = "basic",
                  reportCallIssue = "none",
                  disableOrganizeImports = true,
                },
              },
            },
          },
        },
      }
```
